pallets/liquidity-mining/src/lib.rs

```diff
@@ -0,0 +1,1325 @@
+						g_pool.total_shares_z,
+						g_pool.max_reward_per_period,
+					)?;
+					Self::update_global_pool(g_pool, now_period, reward_per_period)?;
+
+					let pool_reward = Self::claim_from_global_pool(g_pool, liq_pool, liq_pool.stake_in_global_pool)?;
+					Self::update_liq_pool(liq_pool, pool_reward, now_period, g_pool.id, g_pool.reward_currency)?;
+
+					let incentivized_token_balance_in_amm =
+						T::MultiCurrency::free_balance(g_pool.reward_currency, &amm_pool_id);
+
+					let new_stake_in_global_pool = incentivized_token_balance_in_amm
+						.checked_mul(liq_pool.total_valued_shares)
+						.ok_or(Error::<T>::Overflow)?
+						.checked_mul(multiplier.into())
+						.ok_or(Error::<T>::Overflow)?;
+
+					g_pool
+						.total_shares_z
+						.checked_sub(liq_pool.stake_in_global_pool)
+						.ok_or(Error::<T>::Overflow)?;
+					g_pool
+						.total_shares_z
+						.checked_add(new_stake_in_global_pool)
+						.ok_or(Error::<T>::Overflow)?;
+
+					liq_pool.stake_in_global_pool = new_stake_in_global_pool;
+					liq_pool.multiplier = multiplier;
+
+					Ok(())
+				})
+			})
+		}
+
+		#[pallet::weight(1000)]
+		#[transactional]
+		pub fn cancel_liquidity_pool(origin: OriginFor<T>, farm_id: PoolId, asset_pair: AssetPair) -> DispatchResult {
+			let who = ensure_signed(origin)?;
+
+			let amm_account = T::AMM::get_pair_id(asset_pair);
+
+			<LiquidityPoolData<T>>::try_mutate(farm_id, amm_account, |maybe_liq_pool| {
+				let liq_pool = maybe_liq_pool.as_mut().ok_or(Error::<T>::LiquidityPoolNotFound)?;
+
+				ensure!(!liq_pool.canceled, Error::<T>::LiquidityMiningCanceled);
+
+				<GlobalPoolData<T>>::try_mutate(farm_id, |maybe_g_pool| {
+					let g_pool = maybe_g_pool.as_mut().ok_or(Error::<T>::FarmNotFound)?;
+
+					ensure!(g_pool.owner == who, Error::<T>::Forbidden);
+
+					let now_period = Self::get_now_period(g_pool.blocks_per_period)?;
+					Self::maybe_update_pools(g_pool, liq_pool, now_period)?;
+
+					g_pool.total_shares_z = g_pool
+						.total_shares_z
+						.checked_sub(liq_pool.stake_in_global_pool)
+						.ok_or(Error::<T>::Overflow)?;
+
+					liq_pool.canceled = true;
+					liq_pool.stake_in_global_pool = 0;
+
+					Self::deposit_event(Event::LiquidityMiningCanceled(who, farm_id, liq_pool.id, asset_pair));
+
+					Ok(())
+				})
+			})
+		}
+
+		#[pallet::weight(1000)]
+		#[transactional]
+		pub fn remove_liquidity_pool(
+			origin: OriginFor<T>,
+			farm_id: PoolId,
+			asset_pair: AssetPair,
+		) -> DispatchResultWithPostInfo {
+			let who = ensure_signed(origin)?;
+
+			let amm_account = T::AMM::get_pair_id(asset_pair);
+
+			<LiquidityPoolData<T>>::try_mutate_exists(
+				farm_id,
+				amm_account,
+				|maybe_liq_pool| -> DispatchResultWithPostInfo {
+					let liq_pool = maybe_liq_pool.as_mut().ok_or(Error::<T>::LiquidityPoolNotFound)?;
+
+					ensure!(liq_pool.canceled, Error::<T>::LiquidityMiningIsNotCanceled);
+
+					<GlobalPoolData<T>>::try_mutate(farm_id, |maybe_g_pool| -> DispatchResultWithPostInfo {
```

Comment:
You can use DispatchResult instead of DispatchWithPostInfo, no ? Like you do everywhere else ?